engine/src/main/java/org/camunda/bpm/engine/impl/ProcessEngineImpl.java

```diff
@@ -149,7 +151,15 @@ protected void executeSchemaOperations() {
     try {
       commandExecutorSchemaOperations.execute(processEngineConfiguration.getProcessEngineBootstrapCommand());
     } catch (OptimisticLockingException ole) {
+      // if an OLE occurred during the process engine bootstrap, we suppress it
+      // since all the data has already been persisted by a previous process engine bootstrap
       LOG.historyCleanupJobReconfigurationFailure(ole);
+      String databaseType = this.getProcessEngineConfiguration().getDatabaseType();
+      if (DbSqlSessionFactory.CRDB.equals(databaseType)) {
+        // on CRDB, we want to re-throw the OLE to the caller
+        // when the CRDB Command retries are exausted
+        throw ole;
```

Comment:
Why do we want to throw the exception in case of CockroachDB?

Reply:
In the regular case, the catch block suppresses the OLE since the changes made by the concurrent transaction are acceptable, and the conflict can be ignored so that the Process Engine bootstrap can complete successfully.

In the CRDB case, if the `BootstrapEngineCmd` exausts the maximum number of `commandRetries`, we want to notify the caller that the Process Engine bootstrap failed since the transaction was aborted/rolled-back and must be manually retried.